Bowling/Bowling/Frames.cs

```diff
@@ -0,0 +1,177 @@
+﻿using NUnit.Framework.Constraints;
+using NUnit.Framework.Internal.Execution;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace BowlingApp
+{
+    public class Frames
+    {
+        public const int frameNumberTotal = 10;
+        public const int frameNumberWithBonusRoll = 10;
+        private int placeCount = 0;
+        private string score = string.Empty;
+
+        public List<Frame> CreateAllFrames(string bowlingMarks)
+        {
+            List<Frame> framesCollection = new List<Frame>();
+            var marksLength = bowlingMarks.Length;
+
+            try
+            {
+                for (var frameNumber = 1; frameNumber <= frameNumberTotal; frameNumber++)
+                {
+                    if ((placeCount <= marksLength) && (frameNumber != frameNumberWithBonusRoll))
+                    {
+                        framesCollection = CreateFrameNoBonusRoll(frameNumber, bowlingMarks, framesCollection);
+                    }
+                    else
+                    {
+                        framesCollection = CreateFrameWithBonusRoll(frameNumber, bowlingMarks, marksLength, framesCollection);
+                    }
+                }
+            }
+            catch (Exception ex)
```

Comment:
Well you got me there Drew!  I'm not necessarily expecting an error but since we're not doing any real validation on the input I just made this a catch-all for crashing just a tad more gracefully.  Is it your opinion that one isn't needed or that I should be looking for specific exceptions in that statement?

Reply:
I just looked through the instructions for this kata again and it states: "We will not check for valid rolls."

So, in not over-complicating this kata, I think it's safe to assume we'll be given valid input.  Lets remove this `try`-`catch` block and forego any of my other comments about potential ways to trick the program with bad input.